### PR TITLE
Added ability to create invitation without mail send

### PR DESCRIPTION
**Pull request by mingazhev**
We need a flow where invitation link is generated without sending e-mail. 
This PR is about that + a little refactoring

### CODE REVIEW

File: Api/Services/Customers/CustomerInvitationService.cs [hunk 1]
```diff
@@ -29,13 +28,11 @@ public class CustomerInvitationService : ICustomerInvitationService
         }
 
 
-        public async Task<Result> SendInvitation(CustomerInvitationInfo invitationInfo)
+        public async Task<Result> Send(CustomerInvitationInfo invitationInfo)
         {
-            var (_, isFailure, customerInfo, error) = await _customerContext.GetCustomerInfo();
-            if (isFailure)
-                return Result.Fail(error);
+            var (_, customerCompanyId, _, _) = await _customerContext.GetCustomer();
```
Comment (kirillta) [inline Api/Services/Customers/CustomerInvitationService.cs]: Why we got companyId from GetCustomer method?

File: Api/Services/Users/UserInvitationService.cs [hunk 2]
```diff
@@ -130,6 +120,38 @@ private async Task<Maybe<UserInvitation>> GetInvitation(string code)
...
+                InvitationType = invitationType
+            });
+
+            return _context.SaveChangesAsync();
+        }
+        
+        
+        private void LogInvitationCreated(string email)
+            => _logger.LogInvitationCreatedInformation(
+                $"Invitation for user {email} created");
```
Comment (kirillta) [inline Api/Services/Users/UserInvitationService.cs]: "The invitation created for the user {email}"

File: Api/Services/Customers/CustomerInvitationService.cs [hunk 3]
```diff
@@ -54,9 +51,20 @@ public async Task<Result> SendInvitation(CustomerInvitationInfo invitationInfo)
...
                 _options.MailTemplateId, UserInvitationTypes.Customer);
         }
+        
+        
+        public async Task<Result<string>> Create(CustomerInvitationInfo invitationInfo)
+        {
+            var (_, customerCompanyId, _, _) = await _customerContext.GetCustomer();
+
+            if (customerCompanyId != invitationInfo.CompanyId)
+                return Result.Fail<string>("Invitations can be sent within a company only");
```
Comment (kirillta) [inline Api/Services/Customers/CustomerInvitationService.cs]: send

File: Api/Services/Customers/CustomerInvitationService.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Customers/CustomerInvitationService.cs]: It's deconstruction of CustomerInfo

Comment (mingazhev) [reply Api/Services/Customers/CustomerInvitationService.cs]: anyway, rewritten that to more clear

**Pull request merged**